src/pages/search/advanced_search/AdvancedSearchPage.tsx

```diff
@@ -85,7 +85,7 @@ export const AdvancedSearchPage = () => {
     sort: params.get(ResultParam.Sort) as SortOrder | null,
     title: params.get(ResultParam.Title),
     excludeSubunits,
-    topLevelOrganization: excludeSubunits && !!unitId ? null : topLevelOrganizationId,
+    topLevelOrganization: (excludeSubunits && !!unitId) || !!unitId ? null : topLevelOrganizationId,
```

Comment:
```suggestion
    topLevelOrganization: topLevelOrganizationId,
```

burde ikke denne heller vært i `fetchResults`, sånn at det gjelder overalt?
```
if (params.topLevelOrganization) {
  if (params.unit) {
    searchParams.set(ResultParam.Unit, params.unit);
  } else {
    searchParams.set(ResultParam.TopLevelOrganization, encodeURIComponent(params.topLevelOrganization));
  }
}
```

Reply:
Gjorde det litt annerledes, mon tror det skal funke på samme måte. Med mindre det er en case hvor vi faktisk vil ha topLevelOrg og Unit?